contracts/ContinuousTokenFactory.sol

```diff
@@ -2,9 +2,193 @@
+    (ERC20Managed token, Curve curve) = unpack(id);
+
+    // Check price
+    uint256 price = curve.price(token.totalSupply(), amount);
+    require(price <= maxPrice, "ContinuousTokenFactory: price greater than maxPrice");
+
+    // Transfer and mint
+    require(
+      reserveToken.transferFrom(buyer, address(this), price)
+    );
+    token.mint(buyer, amount);
+
+    emit Buy(
+      id,
+      token.name(),
+      amount,
+      price,
+      buyer
+    );
+    return true;
+  }
+
+  event Sell(
+    bytes32 id,
+    string name,
+    uint256 amount,
+    uint256 price,
+    address seller
+  );
+
+  function sell(
+    bytes32 id,
+    uint256 amount,
+    uint256 minPrice,
+    address seller
+  ) external returns (bool) {
+    (ERC20Managed token, Curve curve) = unpack(id);
+    require(token.balanceOf(seller) >= amount, "ContinuousTokenFactory: seller holds too few tokens");
+
+    // Check price
+    uint256 price = curve.price(token.totalSupply().sub(amount), amount);
+    require(price >= minPrice, "ContinuousTokenFactory: price less than minPrice");
+
+    // Burn and transfer
+    token.burn(seller, amount);
+    require(
+      reserveToken.transfer(seller, price)
+    );
+
+    emit Sell(
+      id,
+      token.name(),
+      amount,
+      price,
+      seller
+    );
+    return true;
+  }
+
+  // TODO portals to other ERC20 interface functions
+
+  function totalSupply(bytes32 id) external view returns (uint256) {
+    (ERC20Managed token, ) = unpack(id);
+    return token.totalSupply();
+  }
+
+  function balanceOf(bytes32 id, address account) external view returns (uint256) {
+    (ERC20Managed token, ) = unpack(id);
+    return token.balanceOf(account);
+  }
+
+  function approve(
+    bytes32 id, 
+    address spender, 
+    uint256 amount
+  ) external returns (bool) {
+    (ERC20Managed token, ) = unpack(id);
+    return token.approve(spender, amount);
+  }
+
+  function allowance(
+    bytes32 id,
+    address owner, 
+    address spender
+  ) external view returns (uint256) {
+    (ERC20Managed token, ) = unpack(id);
+    return token.allowance(owner, spender);
+  }
+
+  function transfer(
+    bytes32 id, 
+    address recipient, 
+    uint256 amount
+  ) external returns (bool) {
+    (ERC20Managed token, ) = unpack(id);
+    return token.transfer(recipient, amount);
+  }
+
+  function transferFrom(
+    bytes32 id,
+    address sender, 
+    address recipient, 
+    uint256 amount
+  ) external returns (bool) {
+    (ERC20Managed token, ) = unpack(id);
+    return token.transferFrom(sender, recipient, amount);
+  }
+
+  function exists(bytes32 id) internal view returns (bool) {
+    return continuousTokens[id].token != address(0);
+  }
+
+  function unpack(bytes32 id) internal view returns (ERC20Managed, Curve) {
+    require(exists(id), "ContinuousTokenFactory: invalid id");
+    ContinuousToken storage ct = continuousTokens[id];
+    require(ct.token != address(0), "ContinuousTokenFactory: invalid token");
```

Comment:
I think that these requirements (other than `exist`) should only be checked "on the way in". If we've created an invalid continuous token I think this function shouldn't hide it.